plan/expression_rewriter.go

```diff
@@ -781,6 +781,13 @@ func (er *expressionRewriter) rewriteVariable(v *ast.VariableExpr) {
 	}
 	var val string
 	var err error
+	if v.ExplicitScope {
```

Comment:
Why should we care if it is set explicitly or not?

Reply:
if the scope constraint is not set explicitly, the behavior is to choose the right scope automatically. So the scope check should't happen here, logically.  For example:
for a `global` variable `max_connections`, `select @@max_connections` works, and for a `session` variable `warning_count`, `select @@warnging_count` works as well.